watertap/unit_models/Bipolar_Electrodialysis_0D.py

```diff
@@ -0,0 +1,2343 @@
+        )
+
+        # Performance metrics
+        self.power_electrical = Var(
+            self.flowsheet().time,
+            initialize=1,
+            bounds=(0, 12100),
+            domain=NonNegativeReals,
+            units=pyunits.watt,
+            doc="Electrical power consumption of a stack",
+        )
+        self.specific_power_electrical = Var(
+            self.flowsheet().time,
+            initialize=10,
+            bounds=(0, 1000),
+            domain=NonNegativeReals,
+            units=pyunits.kW * pyunits.hour * pyunits.meter**-3,
+            doc="acidic-volume-flow-rate-specific electrical power consumption",
+        )
+        self.acid_produced = Var(
+            initialize=55 * 1e3,
+            bounds=(0, 1e6),
+            units=pyunits.kg * pyunits.second**-1,
+            doc="Acid prodcued",
+        )
+        self.base_produced = Var(
+            initialize=55 * 1e3,
+            bounds=(0, 1e6),
+            units=pyunits.kg * pyunits.second**-1,
+            doc="Base prodcued",
+        )
+        self.velocity_basic = Var(
+            self.flowsheet().time,
+            initialize=0.01,
+            units=pyunits.meter * pyunits.second**-1,
+            doc="Linear velocity of flow in the base channel of the bipolar membrane",
+        )
+        self.velocity_acidic = Var(
+            self.flowsheet().time,
+            initialize=0.01,
+            units=pyunits.meter * pyunits.second**-1,
+            doc="Linear velocity of flow in the acid channel of the bipolar membrane",
+        )
+        # Parameters for bipolar membrane operation
+        self.elec_field_non_dim = Var(
+            self.flowsheet().time,
+            initialize=1,
+            # bounds=(0, 1e32),
+            units=pyunits.dimensionless,
+            doc="Limiting current density across the bipolar membrane as a function of the normalized length",
+        )
+        self.relative_permittivity = Var(
+            self.membrane_set,
+            initialize=30,
+            bounds=(1, 80),
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            doc="Relative permittivity",
+        )
+        self.membrane_fixed_charge = Var(
+            self.membrane_set,
+            initialize=1.5e3,
+            bounds=(1e-1, 1e5),
+            units=pyunits.mole * pyunits.meter**-3,
+            doc="Membrane fixed charge",
+        )
+        self.kr = Var(
+            self.membrane_set,
+            initialize=1.33 * 10**11,
+            bounds=(1e-6, 1e16),
+            units=pyunits.L * pyunits.mole**-1 * pyunits.second**-1,
+            doc="Re-association rate constant",
+        )
+        self.k2_zero = Var(
+            self.membrane_set,
+            initialize=2 * 10**-5,
+            bounds=(1e-10, 1e2),
+            units=pyunits.second**-1,
+            doc="Dissociation rate constant at no electric field",
+        )
+        self.salt_conc_aem = Var(
```

Comment:
Having some variables denoted as _bpem and some as either _cem or _aem is a little challenging to track how the bpem is parameterized.